external/TBinf/TBinf.cpp

```diff
@@ -68,6 +68,26 @@ void SummaryWriter::add_scalar(const std::string tag, float value,
   write_summary_event(s, step);
 }
 
+void SummaryWriter::add_image(const std::string& tag,
+                              std::string encoded_img,
+                              const std::vector<size_t>& dims,
+                              int64_t step){
+
+  auto s = std::unique_ptr<tensorflow::Summary>(new tensorflow::Summary());
```

Comment:
The `unique_ptr` is the only real option for ensuring basic exception safety. Other than the move ctor/assign operator, none of the generated protobuf API is declared `noexcept`. It looks silly given the current API to `SummaryWriter`, but I'd argue its the only real option (I also would argue that exceptions should be meaningfully catchable and libraries should not make assumptions about what happens to exceptions they emit other than to clean up after themselves in case the downstream can recover).

Reply:
That's a fair point, but in general, I believe Protobuf does not throw exceptions (per the Google style guide). I think more recent versions do throw exceptions in unusual circumstances, but they should not be recovered from.